graphrag/index/flows/create_final_relationships.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2024 Microsoft Corporation.
+# Licensed under the MIT License
+
+"""All the steps to transform final relationships."""
+
+from typing import Any, cast
+
+import pandas as pd
+from datashaper import (
+    VerbCallbacks,
+)
+
+from graphrag.index.cache import PipelineCache
+from graphrag.index.verbs.graph.compute_edge_combined_degree import (
+    compute_edge_combined_degree_df,
+)
+from graphrag.index.verbs.graph.unpack import unpack_graph_df
+from graphrag.index.verbs.text.embed.text_embed import text_embed_df
+
+
+async def create_final_relationships(
+    entity_graph: pd.DataFrame,
+    nodes: pd.DataFrame,
+    callbacks: VerbCallbacks,
+    cache: PipelineCache,
+    text_embed: dict | None = None,
+) -> pd.DataFrame:
+    """All the steps to transform final relationships."""
+    graph_edges = unpack_graph_df(entity_graph, callbacks, "clustered_graph", "edges")
+
+    graph_edges.rename(columns={"source_id": "text_unit_ids"}, inplace=True)
+
+    filtered = cast(
+        pd.DataFrame, graph_edges[graph_edges["level"] == 0].reset_index(drop=True)
+    )
+
+    if text_embed:
+        filtered["description_embedding"] = await text_embed_df(
+            filtered,
+            callbacks,
+            cache,
+            column="description",
+            strategy=text_embed["strategy"],
+            embedding_name="relationship_description",
+        )
+
+    pruned_edges = filtered.drop(columns=["level"])
+
+    filtered_nodes = cast(
+        pd.DataFrame,
+        nodes[nodes["level"] == 0].reset_index(drop=True)[["title", "degree"]],
```

Comment:
this looks weird to me haha, I'd split this in two lines:

```py
filtered_nodes = cast(pd.DataFrame, nodes[nodes["level"] == 0])
filtered_nodes = filtered_nodes[["title", "degree"]].reset_index(drop=True)
```

Reply:
Soooo, I wouldn't sweat too much about pandas optimizations as I'll be sweeping through this as soon as Nathan gives me green light